lib/maths/unittest/CDataFrameUtilsTest.cc

```diff
@@ -34,6 +35,30 @@ using TMeanAccumulator = maths::CBasicStatistics::SSampleMean<double>::TAccumula
 using TMeanAccumulatorVec = std::vector<TMeanAccumulator>;
 using TMeanAccumulatorVecVec = std::vector<TMeanAccumulatorVec>;
 
+auto generateCategoricalData(test::CRandomNumbers& rng,
+                             std::size_t rows,
+                             std::size_t cols,
+                             TDoubleVec expectedFrequencies) {
+
+    TDoubleVecVec frequencies;
+    rng.generateDirichletSamples(expectedFrequencies, cols, frequencies);
+
+    TDoubleVecVec values(cols);
+    for (std::size_t i = 0; i < frequencies.size(); ++i) {
+        for (std::size_t j = 0; j < frequencies[i].size(); ++j) {
+            std::size_t target{static_cast<std::size_t>(
+                static_cast<double>(rows) * frequencies[i][j] + 0.5)};
+            values[i].resize(values[i].size() + target, static_cast<double>(j));
+        }
+        values[i].resize(rows, values[i].back());
```

Comment:
This is the condition which was not guaranteed, in particular, it was possible that `values[i].size() < rows` before.